### ISSUE TITLE
Rerunning compose on Windows errs that network needs to be recreated

### ISSUE THREAD

**Issue by heaths** [docker/compose#4818]
On Windows with docker-compose 1.11.2 and docker 17.03.1-ce, when I rerun a service (e.g. `docker-compose run` or `docker-compose up` I get an error that the network needs to be recreated because options had changed instead of the container (and I made no changes to the _docker-compose.yml_) joining the network. My compose file looks like:

```yaml
# Copyright (C) Microsoft Corporation. All rights reserved.
# Licensed under the MIT license. See LICENSE.txt in the project root for license information.

version: '2.1'
services:
  test:
    build: .
    volumes:
      - "${OUTDIR:-../../src/VSSetup.PowerShell/bin/Debug}:C:\\Users\\ContainerAdministrator\\Documents\\WindowsPowerShell\\Modules\\VSSetup:ro"
      - ../Instances:C:\ProgramData\Microsoft\VisualStudio\Packages\_Instances:ro
      - ../Tests:C:\Tests
      - C:\VS\Community
      - C:\VS\Professional
      - C:\VS\Enterprise
      - C:\BuildTools
    networks:
      - external
networks:
  external:
    driver: transparent
```

**Comment by heaths** [comment]
FWIW, I found that just using `network_mode: transparent` in my compose file works around the issue by not creating a network in the first place, but given the [documentation](https://docs.docker.com/compose/networking/) I would expect this to work.

**Comment by shin-** [comment]
I believe this is fixed by #4794
